src/DIPS.Xamarin.UI/Controls/Toast/Toaster.cs

```diff
@@ -0,0 +1,370 @@
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace DIPS.Xamarin.UI.Controls.Toast
+{
+    internal enum ToastState
+    {
+        Closed = 0,
+        Closing,
+        Opened
+    }
+
+    /// <summary>
+    ///     Toaster control that would appear on top of the presented view
+    /// </summary>
+    public class Toaster : BindableObject
+    {
+        private Toaster()
+        {
+        }
+
+        private static ToastState ToastState { get; set; }
+        private static Guid? Id { get; set; }
+        private CancellationTokenSource CancellationSource { get; set; } = new CancellationTokenSource();
+
+        /// <summary>
+        /// Get the current instance of the Toaster control
+        /// </summary>
+        public static Toaster Current { get; } = new Toaster();
+
+        private Toast GetToast()
+        {
+            var toast = new Toast();
+
+            toast.SetBinding(Toast.TextProperty, new Binding(nameof(Text), source: this));
+            toast.SetBinding(Toast.FontSizeProperty, new Binding(nameof(FontSize), source: this));
+            toast.SetBinding(Toast.FontFamilyProperty, new Binding(nameof(FontFamily), source: this));
+            toast.SetBinding(Toast.TextColorProperty, new Binding(nameof(TextColor), source: this));
+            toast.SetBinding(Toast.BackgroundColorProperty, new Binding(nameof(BackgroundColor), source: this));
+            toast.SetBinding(Toast.CornerRadiusProperty, new Binding(nameof(CornerRadius), source: this));
+            toast.SetBinding(Toast.PaddingProperty, new Binding(nameof(Padding), source: this));
+            toast.SetBinding(Toast.PositionYProperty, new Binding(nameof(PositionY), source: this));
+            toast.SetBinding(Toast.LineBreakModeProperty, new Binding(nameof(LineBreakMode), source: this));
+            toast.SetBinding(Toast.MaxLinesProperty, new Binding(nameof(MaxLines), source: this));
+
+            return toast;
+        }
+
+        private async Task HideToasterIn(int timeInSeconds)
+        {
+            await Task.Delay(timeInSeconds * 1000, CancellationSource.Token);
+
+            await HideToaster();
+        }
+
+        /// <summary>
+        /// Show the Toaster control
+        /// </summary>
+        /// <param name="toaster">The Toaster view to display</param>
+        /// <returns>A void <c>Task</c></returns>
+        public async Task ShowToaster(View toaster = null)
+        {
+            // check state closed
+            if (ToastState != ToastState.Closed)
+            {
+                CancellationSource.Cancel();
+                CancellationSource = new CancellationTokenSource();
+                await HideToasterIn(HideToastIn);
+                return;
+            }
+
+            ToastState = ToastState.Opened;
+            CancellationSource = new CancellationTokenSource();
+
+            // get current page
+            var currentPage = Application.Current.MainPage.Navigation.NavigationStack.LastOrDefault();
```

Comment:
This means that the consumer needs to be using a `NavigationPage` to be able to use the toast.
Let's enable the consumer to use the toast if he's not using a `NavigationPage` as well.

See my comment [here](https://github.com/DIPSAS/DIPS.Xamarin.UI/issues/196#issuecomment-652205432)

Reply:
Allowed for more types of pages